File: backend-spring-feature/src/main/java/com/apptemplate/api/features/departments/Department.java
package com.apptemplate.api.features.departments;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import com.apptemplate.api.common.audit.AuditEntityListener;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "departments")
@EntityListeners(AuditEntityListener.class)
public class Department {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false, length = 50)
    private String code;

    @Column(length = 500)
    private String description;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: backend/dotnet/feature-architecture/full/src/App.Template.Api/Features/Departments/Department.cs
using App.Template.Api.Common.Entities;
using App.Template.Api.Features.Users;

namespace App.Template.Api.Features.Departments;

public class Department : AuditableEntity
{
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public bool IsActive { get; set; } = true;
    public ICollection<User> Users { get; set; } = new List<User>();
}
